test/CadlRanchProjects.Tests/type-property-nullable.cs

```diff
@@ -285,5 +289,89 @@ public Task Type_Property_Nullable_CollectionsString_patchNull() => Test(async (
             var response = await new NullableClient(host, null).GetCollectionsStringClient().PatchNullAsync(RequestContent.Create(value));
             Assert.AreEqual(204, response.Status);
         });
+
+        [Test]
+        public void RequiredNullableAreNotSettable()
+        {
+            var requiredStringList = TypeAsserts.HasProperty(typeof(StringProperty), nameof(StringProperty.NullableProperty), BindingFlags.Public | BindingFlags.Instance);
+
+            Assert.Null(requiredStringList.SetMethod);
+        }
+
+        [Test]
+        public void RequiredNullableListsSerializedAsNull()
+        {
+            var inputModel = new CollectionsStringProperty("required", null);
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual(JsonValueKind.Null, element.GetProperty("nullableProperty").ValueKind);
+        }
+
+        [Test]
+        public void RequiredNullableListsSerializedEmptyWhenCleared()
+        {
+            var inputModel = new CollectionsStringProperty("required", new List<string> {"1", "2"});
+            inputModel.NullableProperty.Clear();
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual(JsonValueKind.Array, element.GetProperty("nullableProperty").ValueKind);
+            Assert.AreEqual(0, element.GetProperty("nullableProperty").GetArrayLength());
+        }
+
+        [Test]
+        public void RequiredNullablePropertiesSerializeWhenSet()
+        {
+            var inputModel = new CollectionsStringProperty("required", new List<string> { "1", "2" });
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual("1", element.GetProperty("nullableProperty")[0].GetString());
+            Assert.AreEqual("2", element.GetProperty("nullableProperty")[1].GetString());
```

Comment:
missing one assert: the length should be 2

Reply:
fixed